client/payments/data/reducer.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ * Internal dependencies
+ */
+import { ACTION_TYPES } from './action-types';
+import { PluginsState } from './types';
+import { Actions } from './actions';
+
+const reducer = (
+	state: PluginsState = {
+		paymentGateways: [],
+		requesting: {},
+		errors: {},
+	},
+	payload?: Actions
+): PluginsState => {
+	if ( payload && 'type' in payload ) {
+		switch ( payload.type ) {
+			case ACTION_TYPES.GET_PAYMENT_GATEWAYS_REQUEST:
+				return {
+					...state,
+					requesting: {
+						...state.requesting,
+						getPaymentGateways: true,
+					},
+				};
+			case ACTION_TYPES.GET_PAYMENT_GATEWAY_REQUEST:
+				return {
+					...state,
+					requesting: {
+						...state.requesting,
+						getPaymentGateway: true,
+					},
+				};
+			case ACTION_TYPES.GET_PAYMENT_GATEWAYS_SUCCESS:
+				return {
+					...state,
+					paymentGateways: payload.paymentGateways,
+					requesting: {
+						...state.requesting,
+						getPaymentGateways: false,
+					},
+				};
+			case ACTION_TYPES.GET_PAYMENT_GATEWAYS_ERROR:
+				return {
+					...state,
+					errors: {
+						...state.errors,
+						getPaymentGateways: payload.error,
+					},
+					requesting: {
+						...state.requesting,
+						getPaymentGateways: false,
+					},
+				};
+			case ACTION_TYPES.GET_PAYMENT_GATEWAY_ERROR:
+				return {
+					...state,
+					errors: {
+						...state.errors,
+						getPaymentGateway: payload.error,
+					},
+					requesting: {
+						...state.requesting,
+						getPaymentGateway: false,
+					},
+				};
+			case ACTION_TYPES.UPDATE_PAYMENT_GATEWAY_REQUEST:
+				return {
+					...state,
+					requesting: {
+						...state.requesting,
+						updatePaymentGateway: true,
+					},
+				};
+			case ACTION_TYPES.UPDATE_PAYMENT_GATEWAY_SUCCESS:
+				const targetIndex = state.paymentGateways.findIndex(
+					( paymentGateway ) =>
+						paymentGateway.id === payload.paymentGateway.id
+				);
+
+				if ( targetIndex === -1 ) {
+					return {
+						...state,
+						paymentGateways: [
+							...state.paymentGateways,
+							payload.paymentGateway,
+						],
+						requesting: {
+							...state.requesting,
+							[ payload.selector ]: false,
```

Comment:
With the above comment, this can be updated to `updatePaymentGateway`.

Reply:
The reason we have `payload.selector` is because [updatePaymentGatewaySuccess](https://github.com/woocommerce/woocommerce-admin/blob/add/6912-date-store-for-payment-gateways/client/payments/data/actions.ts#L65) is being used by both [updatePaymentGateway](https://github.com/woocommerce/woocommerce-admin/blob/add/6912-date-store-for-payment-gateways/client/payments/data/actions.ts#L99) and [getPaymentGateway](https://github.com/woocommerce/woocommerce-admin/blob/add/6912-date-store-for-payment-gateways/client/payments/data/resolvers.ts#L43).

We can create a new action `getPaymentGatewaySuccess` and update the reducer to `case ACTION_TYPES.UPDATE_PAYMENT_GATEWAY_SUCCESS|GET_PAYMENT_GATEWAY_SUCCESS`, but that's gonna have code duplication. 

I think we can leave it as it is for now. What do you think? Do you prefer adding a new action for the `getPaymentGateway`?

Reply:
I would prefer a new action here.  I know it seems a bit redundant and maybe less DRY, but we reduce the need for this payload selector and I think someone coming into this code may be confused by `updatePaymentGatewaySuccess` being used after `getPaymentGateway` success.

Reply:
@joshuatf sounds good. I will make an update.

Reply:
updated in https://github.com/woocommerce/woocommerce-admin/commit/fda79219d03c31d3ab2fd0fecdcf1dd31ee5b88b